src/data/mystery-encounters/encounters/choice-of-balance-encounter.ts

```diff
@@ -0,0 +1,393 @@
+  LUCK,
+  PLAYER_STATS,
+  ENEMY_STATS,
+  PP,
+  INSTANT_MONEY,
+  INSTANT_CANDY
+}
+
+export function getNegativeRewards(totalOptions: number): number[] {
+  const negativeOptions = [];
+  const numNegatives = Object.keys(NegativeRewards).filter(nr => !isNaN(Number(nr)));
+  while (negativeOptions.length < totalOptions) {
+    const roll = randSeedInt(numNegatives.length);
+    if (!negativeOptions.includes(roll)) {
+      negativeOptions.push(roll);
+    }
+  }
+  return negativeOptions;
+}
+
+export function getPositiveRewards(totalOptions: number): number[] {
+  const positiveOptions = [];
+  const numPositives = Object.values(PositiveRewards).filter(pr => !isNaN(Number(pr)));
+  while (positiveOptions.length < totalOptions) {
+    const roll = randSeedInt(numPositives.length);
+    if (!positiveOptions.includes(numPositives[roll])) {
+      positiveOptions.push(numPositives[roll]);
+    }
+  }
+  return positiveOptions;
+}
+
+export class RewardOption {
+  public negativeOption: number;
+  public positiveOption: number;
+  public negativeStat = -1;
+  public positiveStat = -1;
+  public negativeAura: Aura;
+  public positiveAura: Aura;
+
+  constructor(negativeOption: number, positiveOption: number) {
+    this.negativeOption = negativeOption;
+    this.positiveOption = positiveOption;
+  }
+
+  generateMessage(): string {
+    const [negativeType, negativeText, negativeStrength, negativeDuration] = this.getRewardInfo(this.negativeOption);
+    const [positiveType, positiveText, positiveStrength, positiveDuration] = this.getRewardInfo(this.positiveOption);
+    const outputMessage: string[] = [];
+    outputMessage.push(this.getDescriptorText(negativeText));
+    if (this.formattedStrengths(negativeType, negativeStrength) !== "") {
+      outputMessage.push(this.formattedStrengths(negativeType, negativeStrength));
+    }
+    outputMessage.push(this.getDescriptorText("for"));
+    outputMessage.push(this.formattedWaves(negativeDuration));
+    outputMessage.push(this.getDescriptorText("then"));
+    outputMessage.push(this.getDescriptorText(positiveText));
+    if (this.formattedStrengths(positiveType, positiveStrength) !== "") {
+      outputMessage.push(this.formattedStrengths(positiveType, positiveStrength));
+    }
+    if (positiveDuration !== 0) {
+      outputMessage.push(this.getDescriptorText("for"));
+    }
+    outputMessage.push(this.formattedWaves(positiveDuration));
+    return outputMessage.join(" ");
+  }
+
+  getRewardInfo(reward: number): [index: number, dialogueName: string, auraStrength: number, auraDuration: number] {
+    //if (Object.values(PositiveRewards).includes(reward) || Object.values(NegativeRewards).includes(reward)) {
+    return statMessageIndex.find(r => r[0] === reward);
+    //return statMessageIndex.filter(r => r[0] === reward)[1];
+    //}
+  }
+
+  private formattedStrengths(type: number, strength: number): string {
```

Comment:
I'd recommend moving this function to i18n formatters. Check out this example: https://github.com/AsdarDevelops/PokeRogue-Events/blob/ec49cd0f4a39025bb5323780a04ce14678b27b50/src/plugins/i18n.ts#L142